core/new-gui/src/app/workspace/component/result-panel/result-panel.component.ts

```diff
@@ -224,18 +263,72 @@ export class ResultPanelComponent {
     this.breakpointAction = false;
   }
 
+  /**
+   * Callback function for table query params changed event
+   *   params containing new page index, new page size, and more
+   *   (this function will be called when user switch page)
+   *
+   * @param params new parameters
+   */
+  public onTableQueryParamsChange(params: NzTableQueryParams) {
+    const { pageSize: newPageSize, pageIndex: newPageIndex } = params;
+    this.currentPageSize = newPageSize;
+    this.currentPageIndex = newPageIndex;
+
+    if (this.isFrontPagination) {
+      return;
+    }
+
+    this.isLoadingResult = true;
+    this.workflowWebsocketService.send('ResultPaginationRequest', { pageSize: newPageSize, pageIndex: newPageIndex });
+    this.workflowWebsocketService.websocketEvent().subscribe(websocketEvent => {
+      if (websocketEvent.type !== 'PaginatedResultEvent') {
+        return;
+      }
+
+      const highlightedOperators = this.workflowActionService.getJointGraphWrapper().getCurrentHighlightedOperatorIDs();
+
+      for (const result of websocketEvent.paginatedResults) {
+        if (result.operatorID === highlightedOperators[0]) {
+          this.total = result.totalRowCount;
+          this.currentResult = result.table.slice();
+          this.isLoadingResult = false;
+          return;
+        }
+      }
+    });
+  }
+
   /**
    * Updates all the result table properties based on the execution result,
    *  displays a new data table with a new paginator on the result panel.
    *
-   * @param response
+   * @param resultData rows of the result (may not be all rows if displaying result for workflow completed event)
+   * @param totalRowCount if present, is the total number of rows for the result; otherwise, use length of resultData
    */
-  private setupResultTable(resultData: ReadonlyArray<object>) {
+  private setupResultTable(resultData: ReadonlyArray<object>, totalRowCount?: number) {
 
     if (resultData.length < 1) {
       return;
     }
 
+    // if there is no new result
+    //   then restore the previous paginated result data from session storage
+    if (sessionStorage.getItem('newWorkflowExecuted') === 'false') {
+      this.isFrontPagination = false;
+      this.currentResult = JSON.parse(sessionStorage.getItem('currentResult') ?? '[]');
+      this.currentPageIndex = JSON.parse(sessionStorage.getItem('currentPageIndex') ?? '1');
+      this.currentPageSize = JSON.parse(sessionStorage.getItem('currentPageSize') ?? '10');
+      this.total = JSON.parse(sessionStorage.getItem('total') ?? '0');
+
+      const columnKeys1 = JSON.parse(sessionStorage.getItem('columnKeys') ?? '[]') as string[];
```

Comment:
see the sessionstorage comment above

Reply:
Update in commit https://github.com/Texera/texera/pull/903/commits/d668ef2d6eb6c5ef694c00eeaa984ccf51bd860e